server/src/internalClusterTest/java/org/elasticsearch/monitor/metrics/NodeIndexingMetricsIT.java

```diff
@@ -0,0 +1,164 @@
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.monitor.metrics;
+
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.plugins.PluginsService;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.telemetry.Measurement;
+import org.elasticsearch.telemetry.TestTelemetryPlugin;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0, numClientNodes = 0)
+public class NodeIndexingMetricsIT extends ESIntegTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(TestTelemetryPlugin.class);
+    }
+
+    public void testNodeIndexingMetricsArePublishing() throws Exception {
+
+        final String node = internalCluster().startNode();
+        ensureStableCluster(1);
+
+        final TestTelemetryPlugin plugin = internalCluster().getInstance(PluginsService.class, node)
+            .filterPlugins(TestTelemetryPlugin.class)
+            .findFirst()
+            .orElseThrow();
+
+        plugin.resetMeter();
+
+        assertAcked(prepareCreate("test", 1).get());
+
+        final int docsCount = randomIntBetween(500, 1000);
+        for (int i = 0; i < docsCount; i++) {
+            var indexResponse = client().index(new IndexRequest("test").id("doc_" + i).source(Map.of("key", i, "val", i))).actionGet();
+            // check that all documents were created successfully since metric counters below assume that
+            assertThat(indexResponse.status(), equalTo(RestStatus.CREATED));
+        }
+
+        // simulate async apm `polling` call for metrics
+        plugin.collect();
+
+        assertBusy(() -> {
+            var indexingTotal = getRecordedMetric(plugin::getLongAsyncCounterMeasurement, "es.indexing.docs.total");
+            assertThat(indexingTotal.getLong(), equalTo((long) docsCount));
+
+            var indexingCurrent = getRecordedMetric(plugin::getLongGaugeMeasurement, "es.indexing.docs.current.total");
+            assertThat(indexingCurrent.getLong(), equalTo(0L));
+
+            var indexingFailedTotal = getRecordedMetric(plugin::getLongAsyncCounterMeasurement, "es.indexing.indexing.failed.total");
+            assertThat(indexingFailedTotal.getLong(), equalTo(0L));
+
+            var deletionTotal = getRecordedMetric(plugin::getLongAsyncCounterMeasurement, "es.indexing.deletion.docs.total");
```

Comment:
can we delete some documents to see this counter growing?

Reply:
Sure